src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -78,6 +99,30 @@ def _assess_clusters(self, all_clusters):
                 for value in cluster.spark_conf.values():
                     if "dbfs:/mnt" in value or "/dbfs/mnt" in value:
                         failures.append(f"using DBFS mount in configuration: {value}")
+
+                # Checking if Azure cluster config is present in spark config
+                for key, value in cluster.spark_conf.items():
+                    for conf in AZURE_SP_CLUSTER_CONF:
+                        if re.search(conf, key):
+                            confList += [{key: value}]
+                if confList:
+                    clusterDict['clusterName'] = cluster.cluster_name
+                    clusterDict['clusterSparkConf'] = confList
+                    failures.append(clusterDict)
+
+            # Checking if Azure cluster config is present in cluster policies
+            if cluster.policy_id:
+                policy_defn = self._get_cluster_policy(cluster.policy_id)
+                for key, value in policy_defn.items():
+                    if key == 'definition' or key == 'policy_family_definition_overrides':
+                        for pol in AZURE_CLUSTER_POLICY:
+                            if re.search(pol, str(value)):
+                                confPolicyList += [{key: value}]
+                if confPolicyList:
+                    clusterPolicyDict['clusterName'] = cluster.cluster_name
+                    clusterPolicyDict['clusterSparkConf'] = confPolicyList
+                    failures.append(clusterPolicyDict)
```

Comment:
Failures is a list of strings. No dicts allowed. It'll break downstream queries, that show top failures.

Reply:
I put them as string now. Lets discuss how and where to capture the cluster config details.